hack/observability/opentelemetry/readme.md

```diff
@@ -0,0 +1,119 @@
+# OpenTelemetry Tracing
```

Comment:
what do you think about linking this from the development.md doc so it's more discoverable?

Reply:
Good call. I should have updated the "View Telemetry" section in the book regardless, and now I did and linked to the opentelemetry readme.md. (Not squashed yet.)

Reply:
Also as soon as this merges, I'll open the next PR that actually restores Jaeger all-in-one just to view traces without needing App Insights. So some of the language here will return.

Reply:
I updated this PR to include the opentelemetry-collector LICENSE file for the Helm chart.